community/bolt/src/main/java/org/neo4j/bolt/logging/BoltMessageLoggerImpl.java

```diff
@@ -51,11 +53,21 @@
     public static final AttributeKey<String> CORRELATION_ATTRIBUTE_KEY = AttributeKey.valueOf(
             BOLT_X_CORRELATION_ID_HEADER );
 
+    static Supplier<String> remoteAddressSupplier( Channel channel )
+    {
+        return () -> remoteAddress( channel );
+    }
+
     BoltMessageLoggerImpl( BoltMessageLog messageLog, Channel channel )
     {
+        this( messageLog, channel, remoteAddressSupplier( channel ) );
+    }
+
+    BoltMessageLoggerImpl( BoltMessageLog messageLog, Channel channel, Supplier<String> remoteAddressSupplier )
```

Comment:
I think having supplier of remote address is an overkill, it makes code more complex. We can simply accept a `Channel` and mock it in tests, if needed.

Reply:
I recognise that this is really ugly. But the reason I did this because it is impossible to mock a final method.